FranciscoOestmann/Topic3/src/user/service/main/DbClient.java

```diff
@@ -0,0 +1,116 @@
+package user.service.main;
+
+/**
+ * @author frann
+ *
+ */
+
+import java.sql.Connection;
+import java.util.Scanner;
+
+public class DbClient implements UserService, UserServiceDAO{
+
+	//State
+	UserService ser = new UserServiceImpl();
+	UserServiceDAO conn = new UserServiceDAOImpl();
+	Scanner sc = new Scanner(System.in);
+	private String url = " ";
+	private String userName = " ";
+	private String userPass = " ";
+	private String dbType = " ";
+	public String tableName = " ";
+	
+	//Constructor
+	public DbClient(){
+		setTableName(tableName);
+	}
+	
+
+	//Getters & Setter
+	public String getUrl() {
+		return url;
+	}
+
+	public void setUrl(String url) {
+		this.url = url;
+	}
+
+	public String getUserName() {
+		return userName;
+	}
+
+	public void setUserName(String userName) {
+		this.userName = userName;
+	}
+
+	public String getUserPass() {
+		return userPass;
+	}
+
+	public void setUserPass(String userPass) {
+		this.userPass = userPass;
+	}
+	
+	public String getTableName() {
+		return tableName;
+	}
+
+	public void setTableName(String tableName) {
+		System.out.println("Enter the table name: ");
+		tableName = sc.nextLine();
```

Comment:
tableName se pisa con this.tableName. y se pierde al salir del método., 
ademas el valor original de tableName no se usa. 
este es un ejemplo de cuando puede servir usar final: setTableName(final String tableName) {